core/src/main/java/org/infinispan/configuration/cache/HashConfiguration.java

```diff
@@ -21,9 +21,9 @@
    public static final AttributeDefinition<ConsistentHashFactory> CONSISTENT_HASH_FACTORY  = AttributeDefinition.builder("consistentHashFactory", null, ConsistentHashFactory.class).immutable().build();
    public static final AttributeDefinition<Hash> HASH = AttributeDefinition.builder("hash", (Hash)MurmurHash3.getInstance()).copier(IdentityAttributeCopier.INSTANCE).immutable().build();
    public static final AttributeDefinition<Integer> NUM_OWNERS = AttributeDefinition.builder("numOwners" , 2).immutable().build();
-   // With the default consistent hash factory, this default gives us an even spread for clusters
-   // up to 6 members and the difference between nodes stays under 20% up to 12 members.
-   public static final AttributeDefinition<Integer> NUM_SEGMENTS = AttributeDefinition.builder("numSegments" , 60).immutable().build();
+   // Because it assigns owners randomly, SyncConsistentHashFactory doesn't work very well with a low number
+   // of segments. (With DefaultConsistentHashFactory, 60 segments was ok up to 6 nodes.)
+   public static final AttributeDefinition<Integer> NUM_SEGMENTS = AttributeDefinition.builder("numSegments", 256).immutable().build();
```

Comment:
Should we have a warning checking if the number of segments is appropriate for the number of nodes, telling users that they should tune it accordingly ?